Trappist/src/Promact.Trappist.Repository/TestConduct/TestConductRepository.cs

```diff
@@ -233,11 +238,85 @@ public TestConductRepository(TrappistDbContext dbContext, ITestsRepository testR
 
             return false;
         }
+
+        public async Task<bool> ExecuteCodeSnippetAsync(int attendeeId, TestAnswerAC testAnswer)
+        {
+            var allTestCasePassed = true;
+            var code = testAnswer.Code;
+
+            var testCases = await _dbContext.CodeSnippetQuestionTestCases.Where(x => x.CodeSnippetQuestionId == testAnswer.QuestionId).ToListAsync();
+
+            foreach(var testCase in testCases)
+            {
+                code.Input = testCase.TestCaseInput;
+                var result = await ExecuteCodeAsync(code);
+
+                if (result.Output != testCase.TestCaseOutput)
+                {
+                    allTestCasePassed = false;
+                }
+            }
+            return allTestCasePassed;
+        }
         #endregion
 
         #region Private Method
 
         /// <summary>
+        /// Calls code base simulator to execute the code
+        /// </summary>
+        /// <param name="code">Code object</param>
+        /// <returns>Result object</returns>
+        private async Task<Result> ExecuteCodeAsync(Code code)
+        {
+            var serializedCode = JsonConvert.SerializeObject(code);
+            var body = new StringContent(serializedCode, System.Text.Encoding.UTF8, "application/json");
+            var response = await client.PostAsync(CodeBaseServer, body);
+            var content = response.Content.ReadAsStringAsync();
+            var result = JsonConvert.DeserializeObject<Result>(content.Result);
+
+            return result;
+        }
+
+        /// <summary>
+        /// Assess Code Snippet Question and update the marks in report table
+        /// </summary>
+        /// <param name="attendeeId">Test Attendee Id</param>
+        /// <param name="testAnswer">TestAnswer object</param>
+        private async Task AssessCodeSnippet(int attendeeId, TestAnswerAC testAnswer)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove the unused private method 'AssessCodeSnippet'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://server12:9005/coding_rules#rule_key=csharpsquid%3AS1144)